src/development/troubleshoot.md

```diff
@@ -140,15 +140,24 @@ WARNING: [pool web] server reached max_children setting (2), consider raising it
 
 That indicates that the server is receiving more concurrent requests than it has PHP processes allocated, which means some requests will have to wait until another finishes.  In this example there are 2 PHP processes that can run concurrently.
 
-Platform.sh sets the number of workers based on the available memory of your container and the estimated average memory size of each process.  There are two ways to increase the number of workers:
+The following command will identify the 20 slowest requests in the last hour, which can provide an indication of what code paths to investigate.
 
-* Adjust the [worker sizing hints](/languages/php/fpm.md) for your project.
-* Upgrade your subscription on Platform.sh to get more computing resources. To do so, log into your [account](https://accounts.platform.sh) and edit the project.
+```bash
+grep $(date +%Y-%m-%dT%H --date='-1 hours') /var/log/php.access.log | sort -k 4 -r -n | head -20
+```
 
+If you see that the processing time of certain requests is slow while its CPU usage is high (e.g. taking more than 1000ms while 65% of time spent on CPU), you may consider using a profiler like [Blackfire](/administration/integrations/blackfire.md) to troubleshoot the application performance issue.
```

Comment:
How can a user tell how much of a request is spent on CPU?  I don't know how to determine that quickly myself.  (At least not without using an app-specific tool like Drupal Devel.)

Reply:
Each line on the FPM access log (php.access.log) has a percentage of time spent on running PHP. That's the percentage of time spent running in the PHP zen engine compared to the total time spent on processing the request.

If the percentage is high, that request processing is basically CPU bound.

Reply:
Ah, then that should be explained.  It's not obvious what all of the columns in that log file mean.

Reply:
I would like to add a mention of the php-fpm access log format to either https://docs.platform.sh/languages/php.html or https://docs.platform.sh/languages/php/fpm.html. Or ... we're just doing things like https://docs.platform.sh/languages/php/fpm.html#measuring-php-worker-memory-usage `In the fifth column, you'll see the peak memory usage that occurred while each request was handled` on this page?

Which one should we go ahead?

Reply:
Hm.  That log is generated by FPM, so probably the FPM page.  Which means we then should rename that page to not be sizing-specific, and have a log section at the bottom or something.

Reply:
Added a section to explain the FPM access log